### DISCUSSION TITLE
Best practice adding an application with module library to existing workspace?

### DISCUSSION THREAD

**Discussion by boaks** [zephyrproject-rtos/zephyr#47131] [Q&A]
Background:
I currently work on an advanced example for the [Thingy:91 using CoAP/DTLS CID](https://github.com/boaks/zephyr-coaps-client).
Using the "Toolchain Manager" from the "nRF Connector for Desktop" simplifies and speeds up the installation of the zephyr sdk and tools a lot. It leafs a user with an workspace (I think, it's T2). With that, it's not possible, to

```
west init ...
```

in order to add your own application to that workspace. 
As "dirty workaround", I rename the ".west" folder and do then the "west init ...".

Question:
Is there a intended way to extend an existing T2/T3 workspace with an additional application?

**Comment by yashi** [comment]
I think West is designed to support only one application.  An application in this context means one final binary file you can write to you board.  Zephyr RTOS does not support multiple applications like a general OS, such as Linux or Windows.
If you want, you can either:
- Add another _application_ as a thread, so that your board runs two independent functions
- Share Zephyr RTOS and other modules but create a new workspace for your new application.  This implies that you want each binary out put file for each application.  You share all your code except your application.   A West Workspace IIRC does not support multiple application hosting.

**Reply by yashi** [reply]
> A West Workspace IIRC does not support multiple application hosting.

It turns out it _does_ support multiple apps with "[T3: Forest topology](https://docs.zephyrproject.org/latest/develop/west/workspaces.html#t3-forest-topology)".  It reads:
> Useful for those supporting multiple independent applications or downstream distributions with no “central” repository
